src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_NullChecking.cs

```diff
@@ -63,7 +75,7 @@ private static BoundStatement ConstructIfStatementForParameter(SourceParameterSy
                 paramIsNullCondition = factory.MakeNullCheck(loweredLeft.Syntax, loweredLeft, BinaryOperatorKind.Equal);
             }
             // PROTOTYPE : Make ArgumentNullException
-            BoundThrowStatement throwArgNullStatement = factory.Throw(factory.New(factory.WellKnownType(WellKnownType.System_Exception)));
+            BoundThrowStatement throwArgNullStatement = factory.Throw(factory.New(factory.WellKnownType(WellKnownType.System_ArgumentNullException)));
```

Comment:
This doesn't look right. ArgumentNullException takes a parameter that is the name of the parameter that is null.